tests/FakeItEasy.IntegrationTests/ExternalAssemblyGenerator.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace FakeItEasy.IntegrationTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using FakeItEasy.Tests.TestHelpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+
+    public class ExternalAssemblyGenerator : IDisposable
+    {
+        public ExternalAssemblyGenerator()
+        {
+            string assemblyContent = @"
+namespace FakeItEasy.IntegrationTests.External
+{
+    using System;
+    using FakeItEasy;
+
+    public class GuidValueFormatter : ArgumentValueFormatter<Guid>
+    {
+        protected override string GetStringValue(Guid argumentValue)
+        {
+            return argumentValue.ToString(""B"");
+        }
+    }
+}
+";
+
+            var references = GetFrameworkAssemblyLocations()
+            .Concat(new[]
+            {
+                typeof(A).GetTypeInformation().Assembly.Location
+            })
+            .Select(l => MetadataReference.CreateFromFile(l));
+
+            var compilation = CSharpCompilation.Create(
+                "FakeItEasy.ExtensionPoints.External",
+                syntaxTrees: new[] { CSharpSyntaxTree.ParseText(assemblyContent) },
+                references: references,
+                options: new CSharpCompilationOptions(outputKind: OutputKind.DynamicallyLinkedLibrary));
+
+            this.TempAssemblyDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            Directory.CreateDirectory(this.TempAssemblyDirectory);
+
+            this.AssemblyPath1 = Emit(compilation, this.TempAssemblyDirectory, "1");
+            this.AssemblyPath2 = Emit(compilation, this.TempAssemblyDirectory, "2");

```

Comment:
Too lazy! I'd have to make the directory and then copy.  I was of two minds about this as well. I'll update as you say.